python/chemiscope/adapters.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import warnings
+
+try:
+    import ase
+
+    HAVE_ASE = True
+except ImportError:
+    HAVE_ASE = False
+
+
+def frames_to_json(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return [_ase_to_json(frame) for frame in frames]
+    elif HAVE_ASE and isinstance(frames[0], ase.Atom):
+        raise Exception(
+            "expected ase.Atoms, got ase.Atom. "
+            + "Try passing a list of frames instead of a single frame "
+            + "(`[frame]` instead of `frame`)"
+        )
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def atom_properties(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return _ase_atom_properties(frames)
+    elif HAVE_ASE and isinstance(frames[0], ase.Atom):
+        raise Exception(
+            "expected ase.Atoms, got ase.Atom. "
+            + "Try passing a list of frames instead of a single frame "
+            + "(`[frame]` instead of `frame`)"
+        )
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def structure_properties(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return _ase_structure_properties(frames)
+    elif HAVE_ASE and isinstance(frames[0], ase.Atom):
+        raise Exception(
+            "expected ase.Atoms, got ase.Atom. "
+            + "Try passing a list of frames instead of a single frame "
+            + "(`[frame]` instead of `frame`)"
+        )
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
+
+def _ase_to_json(frame):
+    data = {}
+    data["size"] = len(frame)
+    data["names"] = list(frame.symbols)
+    data["x"] = [float(value) for value in frame.positions[:, 0]]
+    data["y"] = [float(value) for value in frame.positions[:, 1]]
+    data["z"] = [float(value) for value in frame.positions[:, 2]]
+
+    if (frame.cell.lengths() != [0.0, 0.0, 0.0]).all():
+        data["cell"] = list(np.concatenate(frame.cell))
+
+    return data
+
+
+IGNORED_ASE_ARRAYS = ["positions", "numbers"]
```

Comment:
are there any others we should consistently ignore?

Reply:
no, these are the one always created by ASE, anything else represents a real property